spec/requests/sample_api_spec.rb

```diff
@@ -0,0 +1,100 @@
+require 'rails_helper'
+
+RSpec.describe "Sample API", type: :request do
+  create_users
+
+  context 'Joe' do
+    before do
+      sign_in @joe
+    end
+
+    context "with a pre-existing project" do
+      before do
+        # Sample setup
+        project = create(:public_project, users: [@joe])
+        create(:alignment_config, name: AlignmentConfig::DEFAULT_NAME)
+        @sample_params = {
+          client: "web",
+          host_genome_id: 1,
+          input_files_attributes: [
+            { source_type: "local", source: "norg_6__nacc_27__uniform_weight_per_organism__hiseq_reads__v6__R1.fastq.gz", parts: "norg_6__nacc_27__uniform_weight_per_organism__hiseq_reads__v6__R1.fastq.gz" },
+            { source_type: "local", source: "norg_6__nacc_27__uniform_weight_per_organism__hiseq_reads__v6__R2.fastq.gz", parts: "norg_6__nacc_27__uniform_weight_per_organism__hiseq_reads__v6__R2.fastq.gz" },
+          ],
+          length: 2,
+          name: "norg_6__nacc_27__uniform_weight_per_organism__hiseq_reads__v6__17",
+          project_id: project.id,
+          do_not_process: false,
+        }
+
+        @metadata_params = {
+          "norg_6__nacc_27__uniform_weight_per_organism__hiseq_reads__v6__17" => {
+            "sex" => "Female",
+            "age" => 100,
+            "water_control" => "No",
+            "sample_type" => "CSF",
+            "nucleotide_type" => "DNA",
+            "collection_date" => "2020-01",
+            "collection_location_v2" => { "title" => "Santa Barbara, CA", "name" => "Santa Barbara, CA" },
+          },
+        }
+
+        @client_params = "web"
+
+        # don't actually do any remote work
+        allow_any_instance_of(Sample).to receive(:set_presigned_url_for_local_upload).and_return(true)
+      end
+
+      describe "when we create a sample via sample upload flow" do
+        it "should properly add the pipeline_type flag, directed_acyclic_graph, to the sample and pipeline run" do
```

Comment:
Is there a way to split these into four different its?

Reply:
Yep, can do